base/meta.jl

```diff
@@ -223,6 +223,50 @@ function parse(str::AbstractString; raise::Bool=true, depwarn::Bool=true)
     return ex
 end
 
+"""
+    parsefile(filename)
+
+Parse the Julia source file located at `filename` and return a `:toplevel`
+expression that contains all expressions in the file.
+
+If `fix_line_number_nodes` is `true` (default), then all of the line number
+nodes in the result will have the correct filename. If `fix_line_number_nodes`
+is `false`, then some of the line number nodes may have `nothing` as the
+filename.
+
+julia> expr = Meta.parsefile("hello.jl")
+"""
+function parsefile(filename::AbstractString; fix_line_number_nodes::Bool = true)
+    fn = String(filename)
+    expr = Base.parse_input_line(read(filename, String); filename = fn)
+    # At this point, `expr` is a `:toplevel` expression that contains
+    # everything in the source file.
+    # `expr` contains multiple `LineNumberNode`s.
+    # Unfortunately, currently some of the `LineNumberNode`s do
+    # not have a filename, i.e. the value of the `file` field is
+    # `nothing`. If `fix_line_number_nodes` is true (default), we
```

Comment:
How does this happen? Naively it sounds like a bug elsewhere which should be fixed rather than worked around here.

Reply:
I'll look into it.